src/components/InviteForm.js

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import * as firebase from 'firebase';
+import YoutubeVideo from './YoutubeVideo';
+import { FirebaseAuth } from 'react-firebaseui';
+import { auth } from '../firebaseInit';
+
+const inviteStateToPendingInvite = (invite: InviteState) => {
+  return {
+    name: invite.name,
+    videoUrl: invite.videoUrl
+  }
+}
+
+class InviteForm extends React.Component<{}, InviteState> {
+    constructor(props: InviteState) {
+      super(props);
+      this.state = {name: "", videoUrl: "", errorMessage: ""};
+    }
+
+    handleOnChange(event: any) : void {
+      this.setState({ name: event.target.value });
+    }
+
+    setVideoURL(event: any) : void {
+      this.setState({ videoUrl: event.target.value })
+    }
+
+    isValidEmail(email: string) : boolean {
+      return email.indexOf("@gmail.com") != -1;
+    }
+
+    handleOnSubmit(event: any) : void {
+      event.stopPropagation();
+
+      if (!this.isValidEmail(this.state.name)) {
+        this.setState({ errorMessage: "Please enter a valid gmail address" });
+      }
+
+      else if (this.state.videoUrl.length <= 0) {
+        this.setState({ errorMessage: "Please enter a valid url" });
+      }
+
+      else {
+        let pendingInvite = inviteStateToPendingInvite(this.state);
+        // TODO send ajax, report errors if pendingInvite failed
```

Comment:
We can probably remove concept of entering email addresses for initial MVP, current flow of REQUEST_INVITE and TRUST op codes means to invite someone all you need to do is send them a link to your own profile, and then they hit a button (this needs to be built) to send "REQUEST_INVITE" op which will tell them how to sign up (and their profile will be unverified until person they requested from then hits TRUST)

Reply:
sounds good, removed. so the piece that needs to be built for this pr is the "Request Invite Form". What pieces of info need to be collected for this form? And who supplies the video url in this case?

Reply:
Person new to the network supplies the video url, so that they are in control of their own join video. We need to have them first sign in, then (if needed) edit their full name, and add their join video url. Based on the full name we will generate their memberId (mid), which will have form their.full.name$XXXX where XXXX is 4 digit salt generated by something like

```
function getNumberSuffix(len: number) {
    const exclusiveMax = Math.pow(10, len);
    return (Math.floor(Math.random() * exclusiveMax) + exclusiveMax).toString().substring(1);
}
```

Idea being that users should rarely interact with the salt except for security-sensitive operations, eg "You've never sent Yanda Peekaboo any RAHA before. Please enter their 4 digit public pin to confirm it's the right person."

For an example REQUEST_INVITE operation take a look at data in `/operations`